OpenRA.Game/Network/Connection.cs

```diff
@@ -33,6 +33,7 @@ public interface IConnection : IDisposable
 		void SendImmediate(List<byte[]> orders);
 		void SendSync(int frame, byte[] syncData);
 		void Receive(Action<int, byte[]> packetFn);
+		void StartRecording(Func<string> chooseFilename);
```

Comment:
I was trying to stay as close to the old one as possible in that it could be used as any connection type. The old recorder was part of the interface so to allow any interface to record I placed it there